src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/InitializeNrGeneration.java

```diff
@@ -0,0 +1,39 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.evolutionaryAlgorithm;
+
+import se.liu.ida.hefquin.engine.queryplan.LogicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.LogicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.*;
+
+public class InitializeNrGeneration {
+
+    public static int countNumOfOp( final LogicalPlan plan ) {
+        final LogicalOperator lop = plan.getRootOperator();
+
+        int nrGeneration = 1;
+        if ( lop instanceof LogicalOpMultiwayJoin) {
+            int count = plan.numberOfSubPlans();
+            nrGeneration = calcFactorial(count);
+            for ( int i = 0; i < count; i++ ) {
+                nrGeneration = nrGeneration * countNumOfOp( plan.getSubPlan( i ) );
+            }
+        }
+        else if ( lop instanceof LogicalOpMultiwayUnion) {
+            int count = plan.numberOfSubPlans();
+            for ( int i = 0; i < count; i++ ) {
+                nrGeneration = nrGeneration * countNumOfOp( plan.getSubPlan( i ) );
+            }
+        }
+        else if ( lop instanceof LogicalOpRequest) {
+            nrGeneration = 1;
+        }
+        return nrGeneration;
+    }
+
+    protected static int calcFactorial( final int n ) {
+        if ( n <= 1 )
+            return 1;
+        else
+            return n * calcFactorial(n - 1);
+    }
```

Comment:
The maximum value for `int` is 2,147,483,647. The factorial of 12 is 479,001,600, which is smaller than 2,147,483,647, but the factorial of 13 (6,227,020,800) is already greater than that. Therefore, you should include an additional check here: if n>12, return Integer.MAX_VALUE

Similarly, in the calculations above (where you multiply), check if the result of the multiplication has become a negative value and, if so, replace this negative value by Integer.MAX_VALUE